app/models/ott/CategoryAssessment.scala

```diff
@@ -51,7 +52,8 @@ object CategoryAssessment {
       theme            <- ottResponse.themes.find(_.id == assessment.themeId)
       exemptions       <- assessment.exemptions.map(x => buildExemption(x.id, x.exemptionType, ottResponse)).sequence
       themeDescription <- ottResponse.themes.find(_.id == assessment.themeId).map(_.theme)
-    } yield CategoryAssessment(id, theme.category, exemptions, themeDescription)
+      regulationUrl    <- ottResponse.legalAct.find(_.id == assessment.regulationId).map(_.regulationUrl)
+    } yield CategoryAssessment(id, theme.category, exemptions, themeDescription, Some(regulationUrl))
```

Comment:
I think instead of putting Some(regulationUrl) we should just have regulationUrl but have it be optional.
It is currently not optional on [LegalActResponse.scala](https://github.com/hmrc/trader-goods-profiles-frontend/pull/570/files#diff-bb0fbb6fc683487ce2422dafc0b7a6c4fbb9266eee6889f319c525e4d57fddd2) but the api docs say it's optional. I think this is why you've made it option type, but I think the option needs to 'bubble up' from LegalActResponse to here, if that makes sense. Everything else looks good.

```
regulationUrl    <- ottResponse.legalAct.find(_.id == assessment.regulationId).map(_.regulationUrl)
    } yield CategoryAssessment(id, theme.category, exemptions, themeDescription, Some(regulationUrl))
```

Reply:
Ha, yes, I missed the optional on LegalActResponse. Good catch Ewan.

Reply:
Great work all in all